cmd/containerd-shim-runhcs-v1/task_hcs.go

```diff
@@ -596,6 +598,50 @@ func (ht *hcsTask) DumpGuestStacks(ctx context.Context) string {
 	return ""
 }
 
+func (ht *hcsTask) Share(ctx context.Context, req *shimdiag.ShareRequest) error {
+	if ht.host == nil {
+		return errors.New("task is not isolated")
+	}
+	guestReq := guestrequest.GuestRequest{
+		ResourceType: guestrequest.ResourceTypeMappedDirectory,
+		RequestType:  requesttype.Add,
+	}
+	if ht.isWCOW {
+		options := &hcsschema.VirtualSmbShareOptions{
+			NoDirectmap: ht.host.DevicesPhysicallyBacked(),
+		}
+		if req.ReadOnly {
+			options.ReadOnly = true
+			options.CacheIo = true
+			options.ShareRead = true
+			options.ForceLevelIIOplocks = true
+		}
+		guestReq.Settings = &hcsschema.MappedDirectory{
+			ContainerPath: req.UvmPath,
+			HostPathType:  "VirtualSmbShareName",
+			ReadOnly:      req.ReadOnly,
+		}
+		_, err := ht.host.AddVSMB(ctx, req.HostPath, guestReq, options)
+		return err
+	}
+	st, err := os.Stat(req.HostPath)
```

Comment:
maybe break these two cases into help functions? not critical, but improves readability :)

Reply:
I'll add commented to signify the LCOW -> WCOW transition, I agree it's unwieldy. I don't think helpers related to VSMB/Plan9 belong in this package and I don't want to export them from /uvm either